frontend/src/component/common/NewConstraintAccordion/ConstraintsList/EditableConstraintsList.tsx

```diff
@@ -39,6 +40,17 @@ export const EditableConstraintsList = forwardRef<
         },
     }));
 
+    useEffect(() => {
+        if (!constraints.every((constraint) => constraintId in constraint)) {
+            setConstraints(
+                constraints.map((constraint) => ({
+                    [constraintId]: uuidv4(),
+                    ...constraint,
+                })),
+            );
+        }
+    }, [constraints, setConstraints]);
```

Comment:
I'm not ready to make this component only accept constraints with ids yet, so this helps us at least set ids on constraints if they don't have them yet.